src/query/db_accessor.hpp

```diff
@@ -106,8 +108,34 @@ class VertexAccessor final {
 
   storage::Result<bool> AddLabel(storage::LabelId label) { return impl_.AddLabel(label); }
 
+  storage::ResultSchema<bool> AddLabelAndValidate(storage::LabelId label) {
+    if (const auto maybe_violation_error = impl_.GetSchemaValidator()->ValidateLabelUpdate(label);
+        maybe_violation_error) {
+      return {*maybe_violation_error};
+    }
+    auto maybe_error = impl_.AddLabel(label);
+    if (maybe_error.HasValue()) {
+      return *maybe_error;
+    }
+    return {maybe_error.GetError()};
+  }
+
```

Comment:
It is not a problem at all but my question is why not keep this validation here, isn't it better than t the validation happens as soon as possible?

Reply:
This was resolved in conversation between Benjamin and me, and the conclusion is we want to keep this logic in the storage engine for now because of these reasons:
 - query engine might send data that are not part of the schema defined  in the storage engine, because of some bug  especially in the beggining
 - validating in the query engine can be considered and optimization that will be possibly implemented in the later stages